src/printer.ts

```diff
@@ -163,9 +167,10 @@ export class PugPrinter {
 		} else if (this.tokens[0]?.type === 'eos') {
 			return '';
 		}
-		for (let index: number = 0; index < this.tokens.length; index++) {
+		let index: number = 0;
+		while (index < this.tokens.length) {
 			this.currentIndex = index;
-			const token: Token = this.tokens[index];
+			let token: Token = this.tokens[index++];
```

Comment:
Please move this `index++` out of it's usage, so a reader can directly see if first or last the increment was made

So it will be:
```ts
index++;
let token: Token = this.tokens[index];
```
or
```ts
let token: Token = this.tokens[index];
index++;
```

Reply:
ok